encoders/image/CLIPImageEncoder/__init__.py

```diff
@@ -0,0 +1,48 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import numpy as np
+import torch
+
+from jina.executors.decorators import batching, as_ndarray
+from jina.executors.encoders.frameworks import BaseTorchEncoder
+from jina.executors.devices import TorchDevice
+
+class CLIPImageEncoder(BaseTorchEncoder):
+    """
+    Encodes data from a `np.ndarray` of shape `BatchSize x (Channel x Height x Width)` into
+    a `np.ndarray` of shape `Batchsize x EmbeddingDimension`.
+
+    Internally, :class:`CLIPImageEncoder` wraps the `CLIP` modeL from https://github.com/openai/CLIP
+    """
+    def __init__(self, model_name: str ='ViT-B/32',
+                 *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.model_name = model_name
+
+    def post_init(self):
+        """
+        :param model_name: the name of the model. Supported models include ``ViT-B/32`` and ``RN50``.
+        """
+        import clip
+
+        assert self.model_name in clip.available_models(),\
+            f'model_name={self.model_name} not in clip.available_models'
+
+        model, _ = clip.load(self.model_name, self.device)
+        self.model = model
+
+    @batching
+    @as_ndarray
+    def encode(self, data: 'np.ndarray', *args, **kwargs) -> 'np.ndarray':
+
```

Comment:
And would be nice to also have something for this method

Reply:
added